pkg/activator/util/transports.go

```diff
@@ -76,22 +78,23 @@ func (rrt *retryRoundTripper) RoundTrip(r *http.Request) (resp *http.Response, e
 	// so the body can be read multiple times.
 	r.Body = NewRewinder(r.Body)
 
-	attempts := rrt.retryer.Retry(func() bool {
+	attempts := 0
+	wait.ExponentialBackoff(rrt.backoffSettings, func() (bool, error) {
```

Comment:
It could, I kept the semantics as is in that: The code after this checks the err variable and if evaluates that. It implements proper "give up" logic already. Checking for the ErrWaitTimout doesn't add much benefit there, although we might want to rethink the give up logic below (which I haven't done in this PR).